src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Specs/InterceptorInfo.cs

```diff
@@ -0,0 +1,197 @@
+                {
+                    RegisterInterceptor(ref _interceptors_OptionsBuilderExt);
+                }
+                else
+                {
+                    Debug.Assert((MethodsToGen.ServiceCollectionExt_Any & overload) is not 0);
+                    RegisterInterceptor(ref _interceptors_serviceCollectionExt);
+                }
+
+                MethodsToGen |= overload;
+
+                void RegisterInterceptor(ref List<InvocationLocationInfo>? infoList) =>
+                    (infoList ??= new()).Add(new InvocationLocationInfo(overload, operation));
+            }
+
+            public InterceptorInfo ToIncrementalValue() =>
+                new InterceptorInfo
+                {
+                    MethodsToGen = MethodsToGen,
+
+                    ConfigBinder = _interceptors_configBinder?.OrderBy(i => i.FilePath).ToImmutableEquatableArray(),
+                    OptionsBuilderExt = _interceptors_OptionsBuilderExt?.OrderBy(i => i.FilePath).ToImmutableEquatableArray(),
+                    ServiceCollectionExt = _interceptors_serviceCollectionExt?.OrderBy(i => i.FilePath).ToImmutableEquatableArray(),
+
+                    ConfigBinder_Bind_instance = _configBinder_InfoBuilder_Bind_instance?.ToIncrementalValue(),
+                    ConfigBinder_Bind_instance_BinderOptions = _configBinder_InfoBuilder_Bind_instance_BinderOptions?.ToIncrementalValue(),
+                    ConfigBinder_Bind_key_instance = _configBinder_InfoBuilder_Bind_key_instance?.ToIncrementalValue(),
+                };
+        }
+    }
+
+    internal sealed class TypeInterceptorInfoBuildler
+    {
+        private readonly Dictionary<ComplexTypeSpec, TypedInterceptorInvocationInfo.Builder> _invocationInfoBuilderCache = new();
+
+        public void RegisterInterceptor(MethodsToGen overload, ComplexTypeSpec type, IInvocationOperation invocation)
+        {
+            if (!_invocationInfoBuilderCache.TryGetValue(type, out TypedInterceptorInvocationInfo.Builder? invocationInfoBuilder))
+            {
+                _invocationInfoBuilderCache[type] = invocationInfoBuilder = new TypedInterceptorInvocationInfo.Builder(overload, type);
+            }
+
+            invocationInfoBuilder.RegisterInvocation(invocation);
+        }
+
+        public ImmutableEquatableArray<TypedInterceptorInvocationInfo>? ToIncrementalValue() =>
+            _invocationInfoBuilderCache.Values
+            .Select(b => b.ToIncrementalValue())
+            .OrderBy(i => i.TargetType.AssemblyQualifiedName)
+            .ToImmutableEquatableArray();
+    }
+
+    internal sealed record TypedInterceptorInvocationInfo
+    {
+        public required ComplexTypeSpec TargetType { get; init; }
+        public required ImmutableEquatableArray<InvocationLocationInfo> Locations { get; init; }
+
+        public void Deconstruct(out ComplexTypeSpec targetType, out ImmutableEquatableArray<InvocationLocationInfo> locations) =>
```

Comment:
This should come for free with a `record`. Is there a reason you needed to define this here?